utils/initializeGift.js

```diff
@@ -0,0 +1,76 @@
+const admin = require("firebase-admin");
+require("firebase/firestore");
+const db = require("../config/db");
+const { EVENT } = require("../config/constants");
+const CollectionTypes = require("../utils/types/CollectionTypes");
+
+/**
+ * sets up a gift with gameAccountUUID for both toymaker and giftee
+ * @inner
+ * @param {string} toymakerGameAccountUUID - UUID of the toymaker
+ * @param {string} gifteeGameAccountUUID - UUID of the giftee
+ * @param {bool} isPrimary - Is this the primary/required gift?
+ * @returns {Result}
+ */
+const initializeGift = async (toymakerGameAccountUUID, gifteeGameAccountUUID, isPrimary = false) => {
+  if (!toymakerGameAccountUUID) {
+    return { error: "no toymakerGameAccountUUID set" };
+  }
+
+  if (!gifteeGameAccountUUID) {
+    return { error: "no gifteeGameAccountUUID set" };
+  }
+
+  let eventDoc = db.collection(CollectionTypes.EVENTS).doc(EVENT);
+
+  let gifteeDoc = eventDoc
+    .collection(CollectionTypes.EVENTS__PARTICIPANTS)
+    .doc(gifteeGameAccountUUID);
+
+  let toymakerDoc = eventDoc
+    .collection(CollectionTypes.EVENTS__PARTICIPANTS)
+    .doc(toymakerGameAccountUUID);
+
+  let giftDoc = eventDoc.collection(CollectionTypes.EVENTS__GIFTS).doc();
+
+  let isGiftInitialized = await giftDoc
+    .set({
+      event: eventDoc,
+      initialized: new Date.toISOString(),
+      isPrimary,
+      received: null,
+      sent: null,
+      reported: false,
+      toymaker: toymakerDoc,
```

Comment:
Is there a possibility that these will be promises? Or how exactly do teh references work?

Reply:
A reference is just a pointer to a location there _might_ be a document. We gotta do `await ...toymaker.get()` when using it, as that is a promise and lookup in the database.